main/cmds_test.go

```diff
@@ -1,3 +1,149 @@
 package main
 
+import (
+	"testing"
 
+	"github.com/stretchr/testify/require"
+	"github.com/go-kit/kit/log"
+	"io/ioutil"
+	"path/filepath"
+	"os"
+
+)
+
+func Test_commandsExist(t *testing.T) {
+	// we expect these subcommands to be handled
+	expect := []string{"install", "enable", "disable", "uninstall", "update"}
+	for _, c := range expect {
+		_, ok := cmds[c]
+		if !ok {
+			t.Fatalf("cmd '%s' is not handled", c)
+		}
+	}
+}
+
+func Test_commands_shouldReportStatus(t *testing.T) {
+	// - certain extension invocations are supposed to write 'N.status' files and some do not.
+
+	// these subcommands should NOT report status
+	require.False(t, cmds["install"].shouldReportStatus, "install should not report status")
+	require.False(t, cmds["uninstall"].shouldReportStatus, "uninstall should not report status")
+
+	// these subcommands SHOULD report status
+	require.True(t, cmds["enable"].shouldReportStatus, "enable should report status")
+	require.True(t, cmds["disable"].shouldReportStatus, "disable should report status")
+	require.True(t, cmds["update"].shouldReportStatus, "update should report status")
+}
+
+func Test_checkAndSaveSeqNum_fails(t *testing.T) {
```

Comment:
@mehaag is there a pass case as well for this? Checking if the SeqNum file exists. You can try looking at Mocking the file and test for a success case.

Reply:
The pass case is below the fail case